test/lit/passes/multi-memory-lowering.wast

```diff
@@ -0,0 +1,82 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: wasm-opt %s --enable-multi-memories --multi-memory-lowering --enable-bulk-memory --enable-extended-const -S -o - | filecheck %s
+
+(module
+  (memory $memory1 1)
+  (memory $memory2 2)
+  (data (memory $memory1) (i32.const 0) "a")
+  (data (memory $memory2) (i32.const 1) "123")
+)
+
+;; CHECK:      (type $none_=>_i32 (func (result i32)))
+
+;; CHECK:      (type $i32_=>_i32 (func (param i32) (result i32)))
+
+;; CHECK:      (global $memory2_page_offset (mut i32) (i32.const 1))
+
+;; CHECK:      (memory $combined_memory 3)
+
+;; CHECK:      (data (i32.const 0) "a")
+
+;; CHECK:      (data (i32.add
+;; CHECK-NEXT:  (global.get $memory2_page_offset)
+;; CHECK-NEXT:  (i32.const 1)
+;; CHECK-NEXT: ) "123")
+
+;; CHECK:      (func $custom_memory_size_0 (result i32)
+;; CHECK-NEXT:  (return
+;; CHECK-NEXT:   (global.get $memory2_page_offset)
+;; CHECK-NEXT:  )
+;; CHECK-NEXT: )
+
+;; CHECK:      (func $custom_memory_size_1 (result i32)
```

Comment:
It would be good to add a middle memory so we test all three cases for memory_size.